modules/apps/digital-signature/digital-signature-impl/src/main/java/com/liferay/digital/signature/internal/manager/DSEnvelopeManagerImpl.java

```diff
@@ -127,6 +127,37 @@ public DSEnvelope getDSEnvelope(long groupId, String dsEnvelopeId) {
 			evenlopeJSONObject -> _toDSEnvelope(evenlopeJSONObject), _log);
 	}
 
+	@Override
+	public JSONObject getDSEnvelopesJSONObject(
+		long groupId, String fromDateString, long count, long startPosition) {
+
+		JSONObject jsonObject = _dsHttp.get(
+			groupId,
+			StringBundler.concat(
+				"envelopes?from_date=", fromDateString, "&count=",
+				String.valueOf(count), "&start_position=",
+				String.valueOf(startPosition),
+				"&include=custom_fields,documents,recipients&order=desc"));
+
+		List<DSEnvelope> dsEnvelopes = JSONUtil.toList(
+			jsonObject.getJSONArray("envelopes"),
+			envelopeJSONObject -> _toDSEnvelope(envelopeJSONObject), _log);
+
+		return JSONUtil.put(
+			"endPosition", jsonObject.getString("endPosition")
+		).put(
+			"envelopes",
+			JSONUtil.toJSONArray(
+				dsEnvelopes, dsEnvelope -> _toJSONObject(dsEnvelope), _log)
+		).put(
+			"resultSetSize", jsonObject.getString("resultSetSize")
```

Comment:
I think we could rename some fields from this response, to be similar to Liferay Rest Responses

lastPage: We need to calculate this
page: We need calculate or parse the actual page to response
envelopes -> items
resultSetSize -> pageSize
totalSetSize -> totalCount